.gitignore

```diff
@@ -27,3 +27,6 @@ migrate_working_dir/
 **/doc/api/
 .dart_tool/
 build/
+
+# Don't check in golden failure output.
+**/failures/*.png
```

Comment:
We probably want to ignore everything under a `/falures` directory, but only where goldens might live.

```
test/**/failures/**
test_goldens/**/failures/**
```